src/Symfony/Component/Config/Definition/BaseNode.php

```diff
@@ -182,6 +193,29 @@ public function isRequired()
     }
 
     /**
+     * Checks if this node is deprecated.
+     *
+     * @return bool
+     */
+    public function isDeprecated()
+    {
+        return $this->deprecatedMessage != null;
+    }
+
+    /**
+     * Returns the deprecated message.
+     *
+     * @param string $node the configuration node name
+     * @param string $path the path of the node
+     *
+     * @return bool
+     */
+    public function getDeprecationMessage($node, $path)
+    {
+        return strtr($this->deprecatedMessage, array('%node%' => $node, '%path%' => $path));
```

Comment:
Isn't `sprintf` (or, in your case, if you'd like to pass an array, `vsprintf`) significantly faster than the other string replacement functions? See, for example: https://www.simplemachines.org/community/index.php?topic=175031.0